conans/client/tools/oss.py

```diff
@@ -24,12 +25,38 @@ def args_to_string(args):
 
 def cpu_count(output=None):
     output = default_output(output, 'conans.client.tools.oss.cpu_count')
+
+    def is_running_in_docker():
+        try:
+            cgroup = "/proc/self/cgroup"
+            return (os.path.exists('/.dockerenv') or
```

Comment:
I don't we should explicitly check for docker in that case (there are container systems, like `CRI-O` / `podman`, for instance), if there are cgroups and limits in place, we should respect them anyway, not just in docker

Reply:
yeah, I need to investigate other container services. I only used docker because it's related to the original issue. I'll take a look in others

Reply:
Since we can obtain cpu count by cgroup, I removed docker verification.